### ISSUE TITLE
Parsing of python type annotations for member variables vs. static/class variables

### ISSUE THREAD

**Issue by PlasmaHH** [doxygen/doxygen#10670]
**Describe the bug**
Consider the following code:

```python
class Frobble:
    def __init__(self):
        self._x = "42"
    _x : int
```

This will create a class when instantiated having a member _x but no static attribute (that is, Frobble._x does not exist). Only when you assign something to the class attribute it will start existing. (Also note that type checkers like mypy would flag as an error the assigment of a str to a variable declared as int )

doxygen however documents the above code as both having an untype member _x and a static protected attribute _x of type int.

I would expect that doxygen (because nothing was assigned to the class level _x ) document only the member, and that as type int.

(btw. I noticed that when you do `self._x: int = 42`  doxygen doesn't document that member at all )

Doxygen version 1.10.0 on opensuse tumbleweed

**Comment by doxygen** [comment]
@PlasmaHH Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
